### PR TITLE
160 refresh of docs + fixing of minor errors & typos

### LINKED ISSUE #160: refresh of docs + fixing of minor errors & typos
Refreshing the docs, and fixing a collection of minor issues & typos encountered during @preminger's intake of the userdocs in winter 2023.

### PR DESCRIPTION
**Pull request by preminger**
## Description of the Pull Request (PR):

Refreshing the docs, and fixing a collection of minor issues & typos encountered during @preminger's intake of the userdocs in winter 2023.

## This fixes or addresses the following GitHub issues:

- Fixes #160

### CODE REVIEW

**Comment by preminger** [comment]
Ending this PR after updating only the following files; a separate Issue (and PR) will be opened for the remainder of the documentation.

- introduction.rst
- quick_start.rst
- security.rst
- definition_files.rst
- build_a_container.rst

File: build_a_container.rst [hunk 1]
```diff
@@ -151,53 +152,68 @@ You can do so with the following command.
...
+you are the same user inside the container as on the host machine. Using
+``sudo`` on the host, to acquire root privileges, ensures we can use
+``apt-get`` as root inside the container.
+
+If you aren't able or do not wish to use ``sudo`` when building a
+container, {Singularity} offers several other options: ``--remote``
+builds, a ``--fakeroot`` mode, and limited unprivileged builds using
+``proot``.
+
+Building with ``--remote``
```
Comment (dtrudg) [inline build_a_container.rst]: Not sure about this title change, as what we are describing is a 'remote build', so perhaps the previous title fits better.

File: quick_start.rst [hunk 2]
```diff
@@ -279,15 +280,15 @@ To pass the ``--containall`` option to the ``run`` command and run a
 
 {Singularity} 2.4 introduced the concept of command groups. For
 instance, to list Linux capabilities for a particular user, you would
-use the ``list`` command in the ``capability`` command group like so:
+use the ``list`` command in the ``capability`` command group, like so:
```
Comment (EmmEff) [inline quick_start.rst]: You changed "like so" to "as follows" elsewhere, so should this as be changed to "as follows"?

File: quick_start.rst [hunk 3]
```diff
@@ -339,28 +340,31 @@ commands to download pre-built images from an external resource like the
...
-changed the image will be altered. If reproducibility is a priority for
-you, try building your images from the Container Library.
+.. note::
+
+   Pulling Docker images reduces reproducibility. If you were to pull a Docker
+   image today and then wait six months and pull again, you are not guaranteed
+   to get the same image. If any of the source layers has changed the image will
+   be altered. If reproducibility is a priority for you, try building your
+   images from the `Container Library <https://cloud.sylabs.io/library>`_
+   instead.
```
Comment (EmmEff) [inline quick_start.rst]: This is not quite correct. If you pull a Docker image using "myimage@DIGEST" (as opposed to "myimage:TAG"), you'll always get the same image. Tags "map" to digests. Tags are mutable. Digests are not.

File: build_a_container.rst [hunk 4]
```diff
@@ -207,48 +223,47 @@ Unprivilged ``proot`` builds
...
 
-Builds using ``proot`` to run unprivileged have limitations, as the emulation of
-the root user is not complete. These builds:
+Unprivileged builds that use ``proot`` have limitations, because
+``proot``'s emulation of the root user is not complete. In particular,
+such builds:
 
-- Do not support arch / debootstrap / yum / zypper bootstraps. Use localimage,
-  library, oras, or one of the docker/oci sources.
+- Do not support ``arch`` / ``debootstrap`` / ``yum`` / ``zypper`` bootstraps. Use ``localimage``, ``library``, ``oras``, or one of the ``docker``/``oci`` sources.
```
Comment (dtrudg) [inline build_a_container.rst]: Please wrap this line at 80 chars.

File: quick_start.rst [hunk 5]
```diff
@@ -530,19 +534,17 @@ Arguments to ``run``
...
    $ singularity run library://alpine echo "hello"
-
    hello
 
-Because {Singularity} runscripts are evaluated shell scripts
-arguments can behave slightly differently than in Docker/OCI
-runtimes, if they contain shell code that may be evaluated. To
-replicate Docker/OCI behaviour you may need additional escaping or
-quoting of arguments.
+Because {Singularity} runscripts are evaluated shell scripts arguments can
```
Comment (EmmEff) [inline quick_start.rst]:
```suggestion
Because {Singularity} runscripts are evaluated shell scripts, arguments can
```

I think (?)

**Review by EmmEff — APPROVED** [review]
A couple of minor suggested corrections... if you don't agree, just ignore them :)

File: quick_start.rst [hunk 6]
```diff
@@ -605,9 +617,9 @@ do with normal Linux commands.
 
 .. _build-images-from-scratch:
 
-***************************
- Build images from scratch
-***************************
+*****************************
+ Building images from scratch
```
Comment (dtrudg) [inline quick_start.rst]: Can remove the leading space here.

The most common way of writing an RST heading has the overline/underline *exactly* the same length as the title text.

Having the overline/underline 1 char before, and 1 char after the title test is from an outdated linting/formatting tool that was applied at some point.

File: quick_start.rst [hunk 7]
```diff
@@ -743,18 +757,19 @@ https://www.sylabs.io/contact/
 
 .. _installation-request:
 
-{Singularity} on a shared resource
-----------------------------------
+***********************************
+ {Singularity} on a shared resource
```
Comment (dtrudg) [inline quick_start.rst:782]: As previous r.e. overline/underline/indent.

File: security.rst [hunk 8]
```diff
@@ -147,56 +151,58 @@ additional namespaces and functionality such as seccomp and cgroups.
...
+key or passphrase. The content of the container then remains private,
+even if the SIF file is shared in public.
 
 Encryption and decryption are performed using the Linux kernel's LUKS2
 feature. This is the same technology routinely used for full disk
 encryption. The encrypted container is mounted directly through the
-kernel. Unlike other container formats, an encrypted container is not
-decrypted to disk in order to run it.
+kernel. Unlike other container formats, the encrypted container is run
+without ever decrypted its contents to disk.
```
Comment (dtrudg) [inline security.rst]:
```suggestion
without ever decrypting its contents to disk.
```

**Review by dtrudg — CHANGES_REQUESTED** [review]
Just a couple of very minor things. Thanks for these improvements!

File: build_a_container.rst (hunk 1 above)
Comment (preminger) [reply build_a_container.rst]: Yes, I agree with this. I was initially concerned that starting a header with a double-backticked flag would make the heading stick out & look unlike other headings, but then I saw that other headings in this document do the same thing, and never went back & reverted this change. Thanks!

**Review by dtrudg — APPROVED** [review]

**Pull request merged**